File: ALittle/Module/ALittleIDE/Other/GameLibrary/CEngine/Script/CEngine/UISystem/Plugin/SpringDialog.lua
-- ALittle Generate Lua
module("ALittle", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittle.DisplayLayout, " extends class:ALittle.DisplayLayout is nil")
SpringDialog = Lua.Class(ALittle.DisplayLayout, "ALittle.SpringDialog")

function SpringDialog:Ctor(ctrl_sys)
	___rawset(self, "_is_show", false)
	self.visible = false
end

function SpringDialog.__getter:show()
	return self._is_show
end

function SpringDialog.__setter:show(value)
	if self._is_show == value then
		return
	end
	self._is_show = value
	if self._anti_loop ~= nil then
		self._anti_loop:SetCompleted()
		self._anti_loop:Stop()
		self._anti_loop = nil
	end
	if value then
		self:ShowAnti(1.1, 200, 200)
	else
		self:HideAnti(1.1, 200)
	end
end

function SpringDialog:ShowAnti(scale, scale_big_ms, scale_return_ms)
	if self._show_dialog == nil then
		return
	end
	self.visible = true
	self.alpha = 0
	self._show_dialog.scale_x = 1
	self._show_dialog.scale_y = 1
	self._anti_loop = LoopList()
	local group = LoopGroup()
	group:AddUpdater(LoopLinear(self._show_dialog, "scale_x", scale, scale_big_ms, 1))
	group:AddUpdater(LoopLinear(self._show_dialog, "scale_y", scale, scale_big_ms, 1))
	group:AddUpdater(LoopLinear(self, "alpha", 1, scale_big_ms, 1))
	self._anti_loop:AddUpdater(group)
	group = LoopGroup()
	group:AddUpdater(LoopLinear(self._show_dialog, "scale_x", 1, scale_return_ms, 1))
	group:AddUpdater(LoopLinear(self._show_dialog, "scale_y", 1, scale_return_ms, 1))
	self._anti_loop:AddUpdater(group)
	self._anti_loop:Start()
end

function SpringDialog:HideAnti(scale, scale_big_ms)
	if self._show_dialog == nil then
		return
	end
	self._anti_loop = LoopList()
	local group = LoopGroup()
	group:AddUpdater(LoopLinear(self._show_dialog, "scale_x", scale, scale_big_ms, 1))
	group:AddUpdater(LoopLinear(self._show_dialog, "scale_y", scale, scale_big_ms, 1))
	group:AddUpdater(LoopLinear(self, "alpha", 0, scale_big_ms, 1))
	self._anti_loop:AddUpdater(group)
	self._anti_loop:AddUpdater(LoopAttribute(self, "visible", false, 1))
	self._anti_loop:Start()
end

File: ALittleIDE/Module/ALittleIDE/Other/GameLibrary/CEngine/JSScript/CEngine/UISystem/Plugin/SpringDialog.js
{
if (typeof ALittle === "undefined") window.ALittle = {};


if (ALittle.DisplayLayout === undefined) throw new Error(" extends class:ALittle.DisplayLayout is undefined");
ALittle.SpringDialog = JavaScript.Class(ALittle.DisplayLayout, {
	Ctor : function(ctrl_sys) {
		this._is_show = false;
		this.visible = false;
	},
	get show() {
		return this._is_show;
	},
	set show(value) {
		if (this._is_show === value) {
			return;
		}
		this._is_show = value;
		if (this._anti_loop !== undefined) {
			this._anti_loop.SetCompleted();
			this._anti_loop.Stop();
			this._anti_loop = undefined;
		}
		if (value) {
			this.ShowAnti(1.1, 200, 200);
		} else {
			this.HideAnti(1.1, 200);
		}
	},
	ShowAnti : function(scale, scale_big_ms, scale_return_ms) {
		if (this._show_dialog === undefined) {
			return;
		}
		this.visible = true;
		this.alpha = 0;
		this._show_dialog.scale_x = 1;
		this._show_dialog.scale_y = 1;
		this._anti_loop = ALittle.NewObject(ALittle.LoopList);
		let group = ALittle.NewObject(ALittle.LoopGroup);
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this._show_dialog, "scale_x", scale, scale_big_ms, 1));
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this._show_dialog, "scale_y", scale, scale_big_ms, 1));
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this, "alpha", 1, scale_big_ms, 1));
		this._anti_loop.AddUpdater(group);
		group = ALittle.NewObject(ALittle.LoopGroup);
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this._show_dialog, "scale_x", 1, scale_return_ms, 1));
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this._show_dialog, "scale_y", 1, scale_return_ms, 1));
		this._anti_loop.AddUpdater(group);
		this._anti_loop.Start();
	},
	HideAnti : function(scale, scale_big_ms) {
		if (this._show_dialog === undefined) {
			return;
		}
		this._anti_loop = ALittle.NewObject(ALittle.LoopList);
		let group = ALittle.NewObject(ALittle.LoopGroup);
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this._show_dialog, "scale_x", scale, scale_big_ms, 1));
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this._show_dialog, "scale_y", scale, scale_big_ms, 1));
		group.AddUpdater(ALittle.NewObject(ALittle.LoopLinear, this, "alpha", 0, scale_big_ms, 1));
		this._anti_loop.AddUpdater(group);
		this._anti_loop.AddUpdater(ALittle.NewObject(ALittle.LoopAttribute, this, "visible", false, 1));
		this._anti_loop.Start();
	},
}, "ALittle.SpringDialog");

}